src/main/scala/org/constellation/lb/Manager.scala

```diff
@@ -68,6 +70,15 @@ class Manager(init: NonEmptyList[Addr], config: LoadbalancerConfig)(
                   )
                   .getOrElse(IO.pure(currentStatus))
             )
+            .flatMap { currentNodes =>
+              val session = findSession(hosts)
+              val (sameSessionNodes, otherSessionNodes) = selectSameSessionNodes(session, currentNodes)
+              Applicative[IO].whenA(otherSessionNodes.nonEmpty) {
+                logger.warn(
+                  s"Evicted nodes not in cluster session ${session.getOrElse("unknown")}: ${otherSessionNodes.keySet} "
+                )
+              } >> IO.pure(NonEmptyMap.fromMapUnsafe(sameSessionNodes))
```

Comment:
hmm, is it possible that `sameSessionNodes` would be an empty one? if so, it's indeed unsafe to call `fromMapUnsafe`

Reply:
is not possible, at least it should have the seed host nodes that are the sources of the session used to filter,

Reply:
hm, but it can remove them later on, right? what if non of the peers are in the majority? it may be an empty list then 🤔

Reply:
source nodes are being used for the very first scan. then, they are like every other node. if they would go down, the load balancer may throw an error here 🤔

Reply:
if the node doesn't respond, is kept in the map with value set to None, so the map still won't be empty.

Reply:
yes, but at some point we remove nodes that are below the threshold. please check that flow. otherwise I'm fine

Reply:
I don't see any place where keys are removed form the map (also, any removal operation change the type to SortedMap, which would mean an unsafe  conversion back to NeMap)

Reply:
What happens when the node is below threshold? Don't we remove it? Or we just keep it but does not use it for proxy?

Reply:
That's taken care downstream (but it doesn't actually evict): 

buildClusterStatus(status).flatMap {
                  case (activeHosts, inactiveHosts) =>
                    val clusterHosts = (activeHosts ++ inactiveHosts) // TODO: Add Eviction of inactive hosts